File: src/test/java/com/mpe85/grampa/SuperParser.java
package com.mpe85.grampa;

import com.mpe85.grampa.rule.Rule;

import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;

public class SuperParser extends TestParser {
	
	@Override
	@SuppressFBWarnings("IL_INFINITE_RECURSIVE_LOOP")
	public Rule<String> root() {
		return sequence(character('a'), this.root());
	}
	
}

File: src/test/kotlin/com/mpe85/grampa/SuperParser.kt
package com.mpe85.grampa

import com.mpe85.grampa.rule.Rule
import edu.umd.cs.findbugs.annotations.SuppressFBWarnings

open class SuperParser : TestParser() {

  @SuppressFBWarnings("IL_INFINITE_RECURSIVE_LOOP")
  override fun root(): Rule<String> = sequence(character('a'), this.root())
  
}
